### ISSUE TITLE
[BUG] - Flight logger configurations

### ISSUE THREAD

**Issue by jordanhavard** [invernyx/smartcars-3-bugs#85]
**Describe the bug**
I've gone to central to add the following (formatted for readability) to my flight logger configuration however when I pause my sim, the message isn't coming through
```
    {
      "condition": "{pauseFlag} equals true",
      "message": "Flight paused"
    },
    {
      "condition": "{pauseFlag} equals false",
      "message": "Flight resumed",
      "initialValue": true
    }
```

**To Reproduce**
1. Update logger configuration. 
2. Open SC3, login and start flight
3. Pause flight

**Expected behavior**
Flight log reflects flight logger configuration

**Screenshots**
On the right, you can see the pause flag is set to `true`
![image](https://user-images.githubusercontent.com/67948773/230689800-46e5d059-79c4-4b6e-a98e-7d5646d2ded9.png)

**System Information (please complete the following information):**
 - OS: Win 10 Pro 10.0.19044 Build 19044
 - Community ID: QV
 - Version 0.6.0.1
 - Plugins used: [chat, flight-breakdown, flight-center, flight-tracking (0.6.1), map-plugin, pilot-center]

**Additional context**
My whole configuration is below (formatted for readability)
```
{
  "events": [
    {
      "condition": "{aircraftType} changed or {aircraftType} not_equals ''",
      "message": "Flying {aircraftType}"
    },
    {
      "condition": "{engine1Firing} not_equals false",
      "message": "Engine 1 On"
    },
    {
      "condition": "{engine1Firing} equals false",
      "message": "Engine 1 Off",
      "initialValue": true
    },
    {
      "condition": "{engine2Firing} not_equals false",
      "message": "Engine 2 On"
    },
    {
      "condition": "{engine2Firing} equals false",
      "message": "Engine 2 Off",
      "initialValue": true
    },
    {
      "condition": "{engine3Firing} not_equals false",
      "message": "Engine 3 On"
    },
    {
      "condition": "{engine3Firing} equals false",
      "message": "Engine 3 Off",
      "initialValue": true
    },
    {
      "condition": "{engine4Firing} not_equals false",
      "message": "Engine 4 On"
    },
    {
      "condition": "{engine4Firing} equals false",
      "message": "Engine 4 Off",
      "initialValue": true
    },
    {
      "condition": "{flapsControl} changed",
      "message": "Flaps set to {flapsControl}%"
    },
    {
      "condition": "{fuelTotalQuantityWeight} increased",
      "message": "Fuel increased to {fuelTotalQuantityWeight}",
      "timeout": 1000
    },
    {
      "condition": "{gearControl} not_equals 0",
      "message": "Gear lever lowered",
      "initialValue": true
    },
    {
      "condition": "{gearControl} equals 0",
      "message": "Gear lever raised"
    },
    {
      "condition": "{pauseFlag} equals true",
      "message": "Paused"
    },
    {
      "condition": "{pauseFlag} not_equals true",
      "message": "Unpaused",
      "initialValue": true
    },
    {
      "condition": "{slewMode} equals true",
      "message": "Slew mode entered"
    },
    {
      "condition": "{slewMode} not_equals true",
      "message": "Slew mode left",
      "initialValue": true
    },
    {
      "condition": "{simulationRate} changed",
      "message": "Sim rate changed to {simulationRate}x"
    },
    {
      "condition": "{stallWarning} equals true",
      "message": "Stalling"
    },
    {
      "condition": "{stallWarning} equals true and {ias} decreased",
      "message": "Reached minimum speed of {ias} kts",
      "timeout": 5000
    },
    {
      "condition": "{stallWarning} not_equals true",
      "message": "Stall recovered",
      "initialValue": true
    },
    {
      "condition": "{overspeedWarning} equals true",
      "message": "Overspeeding"
    },
    {
      "condition": "{overspeedWarning} equals true and {ias} increased",
      "message": "Reached maximum speed of {ias} kts",
      "timeout": 5000
    },
    {
      "condition": "{overspeedWarning} not_equals true",
      "message": "No longer overspeeding",
      "initialValue": true
    },
    {
      "condition": "{altitude} less_than 10000 and {ias} greater_than 254 and {ias} increased and {slewMode} equals false",
      "message": "Reached {ias} kts below 10,000 feet",
      "timeout": 5000
    },
    {
      "condition": "{bank} greater_than 30 or {bank} less_than -30",
      "message": "Bank exceeded 30 degrees",
      "timeout": 1000
    },
    {
      "condition": "{pitch} greater_than 20",
      "message": "Pitch exceeded 20 degrees nose up",
      "timeout": 1000
    },
    {
      "condition": "{pitch} less_than -10",
      "message": "Pitch exceeded 10 degrees nose down",
      "timeout": 1000
    },
    {
      "condition": "{phase} equals 'BOARDING'",
      "message": "Now boarding"
    },
    {
      "condition": "{phase} equals 'PUSH_BACK'",
      "message": "Pushing back"
    },
    {
      "condition": "{phase} equals 'TAXI'",
      "message": "Taxiing out"
    },
    {
      "condition": "{phase} equals 'TAKE_OFF'",
      "message": "Taking off"
    },
    {
      "condition": "{phase} equals 'REJECTED_TAKE_OFF'",
      "message": "Takeoff rejected"
    },
    {
      "condition": "{phase} equals 'CLIMB'",
      "message": "Climbing at {ias} kts, {pitch} degrees pitch, {bank} degrees bank"
    },
    {
      "condition": "{phase} equals 'CRUISE'",
      "message": "Cruising at {cruiseAltitude} ft"
    },
    {
      "condition": "{phase} equals 'DESCENT'",
      "message": "Descending with {milesToGo} nm to go"
    },
    {
      "condition": "{phase} equals 'APPROACH'",
      "message": "Approach started at {altitudeAgl} ft AGL with {milesToGo} nm to go"
    },
    {
      "condition": "{phase} equals 'FINAL'",
      "message": "On final at {altitudeAgl} ft with {milesToGo} nm to go at {ias} kts, {pitch} degrees pitch, {bank} degrees bank"
    },
    {
      "condition": "{phase} equals 'LANDED'",
      "message": "Touched down at {landingRate} fpm, {gForceTouchDown}g, {ias} kts, {pitch} degrees pitch, {bank} degrees bank"
    },
    {
      "condition": "{phase} equals 'GO_AROUND'",
      "message": "Going around at {altitudeAgl} ft AGL, {ias} kts, {pitch} degrees pitch, {bank} degrees bank"
    },
    {
      "condition": "{phase} equals 'TAXI_TO_GATE'",
      "message": "Taxiing to gate, landed in {landingDistance} ft"
    },
    {
      "condition": "{pauseFlag} equals true",
      "message": "Flight paused"
    },
    {
      "condition": "{pauseFlag} equals false",
      "message": "Flight resumed",
      "initialValue": true
    }
  ]
}
```

**Comment by turbofandude** [comment]
I just checked and yup, pause isn't logging properly anymore. I will figure out why. Thank you!

**Comment by turbofandude** [comment]
Will be fixed in the next release. For anyone interested, we added a lockout to the main flight logic loop to skip if the time since the last frame was zero. This prevented it from logging erroneous things or passing timeouts (cruise detection, etc.) while paused. That, however, stopped logging logic **before** the pause was logged.

I've changed it now to use our logical elapsedTime value as the time, so it can still run while paused because time isn't advancing even if the functions run, so it doesn't cause issues.
